cypress/e2e/webTables.cy.js

```diff
@@ -1,7 +1,115 @@
 /// <reference types='cypress' />
 
 describe('Web Tables page', () => {
-  it('', () => {
+  beforeEach(() => {
+    cy.visit('/');
+  });
+
+  let employee;
+  before(() => {
+    cy.task('generateEmployee').then((generateEmployee) => {
+      employee = generateEmployee;
+    });
+  });
+
+  it('validate pagination', () => {
+    cy.addNewEmployee(employee);
+    cy.addNewEmployee(employee);
+    cy.addNewEmployee(employee);
+
+    cy.get('[aria-label="rows per page"]').select('5');
+
+    cy.contains('[type="button"]', 'Next').click();
+    cy.get('[type="number"]').should('have.value', '2');
+
+    cy.contains('[type="button"]', 'Previous').click();
+    cy.get('[type="number"]').should('have.value', '1');
+  });
+
+  it('allows rows count selection', () => {
+    cy.get('[aria-label="rows per page"]').should('have.value', '10');
+  });
+
+  it('allows to add a new worker', () => {
+    cy.get('#addNewRecordButton').click();
+    cy.get('.modal-content').should('contain', 'Registration Form');
+
+    cy.findByPlaceholder('First Name').type(employee.firstName);
+    cy.findByPlaceholder('Last Name').type(employee.lastName);
+    cy.findByPlaceholder('name@example.com').type(employee.email);
+    cy.findByPlaceholder('Age').type(employee.age);
+    cy.findByPlaceholder('Salary').type(employee.salary);
+    cy.findByPlaceholder('Department').type(employee.department);
+    cy.get('#submit').click();
+    cy.get('.ReactTable').should('contain', employee.firstName)
+      .and('contain', employee.lastName)
+      .and('contain', employee.email)
+      .and('contain', employee.age)
+      .and('contain', employee.salary)
+      .and('contain', employee.department);
+  });
+
+  it('allows to delete a worker', () => {
+    cy.get('#addNewRecordButton').click();
+    cy.get('.modal-content').should('contain', 'Registration Form');
+
+    cy.findByPlaceholder('First Name').type('Shanti');
+    cy.findByPlaceholder('Last Name').type('Shantihi');
+    cy.findByPlaceholder('name@example.com').type('shanti@gmail.com');
+    cy.findByPlaceholder('Age').type('42');
+    cy.findByPlaceholder('Salary').type('5000');
+    cy.findByPlaceholder('Department').type('Development');
+    cy.get('#submit').click();
+    cy.get('#delete-record-4').click();
+    cy.get('.ReactTable.-striped.-highlight').should('not.contain', 'Shanti')
+      .and('not.contain', 'Shantihi');
+  });
+
+  it('allows to delete all workers', () => {
+    cy.get('#delete-record-1').click();
+    cy.get('#delete-record-2').click();
+    cy.get('#delete-record-3').click();
+    cy.get('.ReactTable.-striped.-highlight')
+      .should('contain', 'No rows found');
+  });
+
+  it('allows to find a worker in the search field and edit it', () => {
```

Comment:
```suggestion
  it('allows to find a worker in the search field and edit it', () => {
```

this test is useless, you do the same things in the test below